src/util/system.cpp

```diff
@@ -1089,11 +1089,12 @@ void AllocateFileRange(FILE *file, unsigned int offset, unsigned int length) {
         fcntl(fileno(file), F_PREALLOCATE, &fst);
     }
     ftruncate(fileno(file), fst.fst_length);
-#elif defined(__linux__)
+#else
+    #if defined(__linux__)
     // Version using posix_fallocate
     off_t nEndPos = (off_t)offset + length;
-    posix_fallocate(fileno(file), 0, nEndPos);
-#else
+    if (!posix_fallocate(fileno(file), 0, nEndPos)) return;
```

Comment:
Might be cleaner to just compare `0 == posix_fallocate(...`

Reply:
Yes, I agree. 0 == ... is more expressive and less error prone.